sched/semaphore/sem_holder.c

```diff
@@ -489,117 +411,52 @@ static int nxsem_restoreholderprio(FAR struct semholder_s *pholder,
       nxsem_freeholder(sem, pholder);
     }
 
+  /* We attempt to restore thread priority to its base priority.  If
+   * there is any thread with the higher priority waiting for the
+   * semaphore held by htcb then this value will be overwritten.
+   */
+
+  hpriority = htcb->boost_priority > htcb->base_priority ?
+              htcb->boost_priority : htcb->base_priority;
+
```

Comment:
Are we boosting (raising) the priority here, or lowering it?

Reply:
Ok. I will extend the comment. This case covers the kwork thread that might have boosted priority.